src/CppAst/CppNoneTypeTemplateParameterType.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Alexandre Mutel. All rights reserved.
+// Licensed under the BSD-Clause 2 license.
+// See license.txt file in the project root for full license information.
+
+using System;
+
+namespace CppAst
+{
+    /// <summary>
+    /// A C++ template parameter type.
+    /// </summary>
+    public sealed class CppNoneTypeTemplateParameterType : CppType
```

Comment:
This naming is confusing, I would believe that it should be better named like `CppTemplateParameterNonType`